src/main/java/de/tum/in/www1/artemis/web/rest/admin/AdminStandardizedCompetencyResource.java

```diff
@@ -107,12 +108,49 @@ public ResponseEntity<Void> deleteStandardizedCompetency(@PathVariable long comp
      */
     @PostMapping("standardized-competencies/knowledge-areas")
     @EnforceAdmin
-    public ResponseEntity<KnowledgeArea> createKnowledgeArea(@RequestBody KnowledgeArea knowledgeArea) throws URISyntaxException {
+    public ResponseEntity<KnowledgeAreaDTO> createKnowledgeArea(@RequestBody KnowledgeAreaDTO knowledgeArea) throws URISyntaxException {
         log.debug("REST request to create knowledge area : {}", knowledgeArea);
 
         var persistedKnowledgeArea = knowledgeAreaService.createKnowledgeArea(knowledgeArea);
 
-        return ResponseEntity.created(new URI("/api/standardized-competencies/knowledge-areas" + persistedKnowledgeArea.getId())).body(persistedKnowledgeArea);
+        return ResponseEntity.created(new URI("/api/standardized-competencies/knowledge-areas" + persistedKnowledgeArea.getId())).body(KnowledgeAreaDTO.of(persistedKnowledgeArea));
+    }
+
+    /**
+     * PUT api/admin/standardized-competencies/knowledge-areas/{knowledgeAreaId} : Updates an existing knowledge area
+     *
+     * @param knowledgeAreaId the id of the knowledge area that should be updated
+     * @param knowledgeArea   the updated knowledge area
+     * @return the ResponseEntity with status 200 (OK) and with body the updated knowledge area
+     */
+    @PutMapping("standardized-competencies/knowledge-areas/{knowledgeAreaId}")
+    @EnforceAdmin
+    public ResponseEntity<KnowledgeAreaDTO> updateKnowledgeArea(@PathVariable long knowledgeAreaId, @RequestBody KnowledgeAreaDTO knowledgeArea) {
+        log.debug("REST request to update knowledge area : {}", knowledgeArea);
+
+        if (knowledgeAreaId != knowledgeArea.id()) {
+            throw new BadRequestException("The knowledge area id in the body and path do not match");
```

Comment:
Why not declaring the id in the DTO as `@JsonProperty(access = JsonProperty.Access.READ_ONLY)` so that the client can not write the id but can read it in the return value. And for the update use the id given in the path and not the id from the dto in the body. (Hopefully it is clear what I mean)